cfme/tests/cloud/test_providers.py

```diff
@@ -36,6 +46,107 @@ def enable_regions(provider):
     enable_provider_regions(provider)
 
 
+@pytest.fixture(scope='function')
+def az_pwsh_vm(provider, appliance, request):
+    # azure_pwsh contains powershell and necessary scripts
+    vm_name = random_vm_name(context="pwsh")
+    template = provider.data.get('powershell_vm')
+    vm = provider.appliance.collections.infra_vms.instantiate(vm_name, provider, template)
+
+    if not provider.exists:
+        provider.create()
+        provider.validate()
+    provisioning_data = {
+        'catalog': {'vm_name': vm_name},
+        'hardware': {'disk_format': 'Default'}}
+
+    do_vm_provisioning(vm_name=vm_name, appliance=appliance, provider=provider,
+                       provisioning_data=provisioning_data, template_name=template,
+                       request=request, num_sec=2500)
+
+    vm.mgmt.ensure_state(VmState.RUNNING)
+
+    @wait_for_decorator(timeout="10m", delay=5)
+    def get_ip_address():
+        ip = vm.mgmt.ip
+        return ip is not None
+    yield vm
+
+    vm.cleanup_on_provider()
+
+
+@pytest.fixture
+def pwsh_ssh(az_pwsh_vm):
+    """Provide vm_ssh_client for ssh operations in the test."""
+    with ssh.SSHClient(hostname=az_pwsh_vm.ip_address,
+                       username=credentials['host_default']['username'],
+                       password=credentials['host_default']['password']) as vm_ssh_client:
+        yield vm_ssh_client
+
+
+@pytest.fixture(scope='function')
+def cfme_vhd(appliance, pwsh_ssh):
+    ip_of_recourse = pwsh_ssh.run_command(
+        'cd /mnt/space && pwsh get_ip.ps1 | grep -oE "([0-9]{1,3}\.){3}[0-9]{1,3}"',
+        timeout=60).output.strip()
+    if ip_of_recourse is not None:
+        pytest.skip("The resource is taken by some other VM in Azure")
+    stream = appliance.version.stream()
+    url = conf.cfme_data['basic_info']['cfme_images_url'][stream]
+    image = pwsh_ssh.run_command(
+        """wget -qO- {url} | grep -Po '(?<=href=")[^"]*' | grep azure""".format(url=url),
+        timeout=30).output.strip()
+    image_url = '{}/{}'.format(url, image)
+    pwsh_ssh.run_command("cd /mnt/space && wget {image}".format(image=image_url), timeout=180)
```

Comment:
Consider defining the first part of this command in a variable and formatting it into the strings.